lib/MusicBrainz/Server/Data/Collection.pm

```diff
@@ -136,33 +136,51 @@ sub get_first_collection {
     return $self->sql->select_single_value($query, $editor_id);
 }
 
-sub find_all_by_editor {
```

Comment:
I combined `find_all_by_editor` and `find_all_by_entity` into a single `find_by` subroutine that takes a hash ref with params, since they shared a lot of code.

Instead of `$show_private` just being 0/1, it's now the id of the editor to include private collections for. This seems safer since we won't accidentally return other private collections we don't care about. It also won't screw up the query limit/offset behavior.

I had to add an `order_by` param to `query_to_list_limited` to get this returning a consistent order. Since `query_to_list_limited` wraps the given query inside a CTE, it loses any internal ordering, so it has to be appended to the outer query.